gobblin-data-management/src/main/java/gobblin/data/management/copy/SubsetFilesCopyableDatasetFinder.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+
+import com.google.common.collect.Lists;
+
+import gobblin.data.management.retention.profile.ConfigurableGlobDatasetFinder;
+import gobblin.util.PathUtils;
+import gobblin.util.filters.RegexPathFilter;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * {@link SubsetFilesCopyableDataset} finder that extends {@link ConfigurableGlobDatasetFinder}.
+ *
+ * It lists files on the root path, and uses a user defined pattern to identify {@link SubsetFilesCopyableDataset}s and
+ * their corresponding subset of files.
+ */
+@Slf4j
+public class SubsetFilesCopyableDatasetFinder extends ConfigurableGlobDatasetFinder<CopyableDataset> {
```

Comment:
It is not recursively finding datasets. The non-recursive listing phase is done in `datasetFinder` rather than `dataset` for performance purpose. Limited logic in `RecursiveCopyableDataset` can be reused here.